test/integration/protocol_integration_test.cc

```diff
@@ -422,6 +422,7 @@ TEST_P(DownstreamProtocolIntegrationTest, RetryHostPredicateFilter) {
 
   // Note how we're expecting each upstream request to hit the same upstream.
   auto upstream_idx = waitForNextUpstreamRequest({0, 1});
+  ASSERT_LT(upstream_idx, fake_upstreams_.size());
```

Comment:
That's the only place where the return value was used.

Reply:
Instead can you have it return an absl::optional? That is more future proof IMO.